File: python-nodes/linear/01_generate_order.py
# Node 1: Generate Order
# Input:  cycle number from previous loop (Node 6) or empty on first run
# Output: a single order object passed to Node 2

import random
import string
from datetime import datetime, timezone

ITEMS = ["Coffee Mug", "Notebook", "USB Cable", "Desk Lamp", "Water Bottle"]

results = []

for item in _items:
    payload = item["json"]
    # Loop-back from webhook wraps body; manual start uses root fields.
    if isinstance(payload.get("body"), dict):
        state = payload["body"]
    else:
        state = payload

    cycle = int(state.get("cycle", 0))

    order = {
        "cycle": cycle,
        "order_id": "ORD-" + "".join(random.choices(string.digits, k=5)),
        "item": random.choice(ITEMS),
        "quantity": random.randint(1, 5),
        "unit_price": round(random.uniform(5.0, 50.0), 2),
        "created_at": datetime.now(timezone.utc).isoformat(),
    }

    results.append({"json": order})

return results

File: javascript-nodes/linear/01_generate_order.js
// Node 1: Generate Order
// Input:  cycle from Node 6 loop or Manual Start
// Output: order object passed to Node 2

const ITEMS = ['Coffee Mug', 'Notebook', 'USB Cable', 'Desk Lamp', 'Water Bottle'];
const results = [];

for (const item of $input.all()) {
  const payload = item.json;
  const state =
    payload.body && typeof payload.body === 'object' ? payload.body : payload;
  const cycle = Number(state.cycle ?? 0);

  results.push({
    json: {
      cycle,
      order_id: 'ORD-' + String(Math.floor(10000 + Math.random() * 90000)),
      item: ITEMS[Math.floor(Math.random() * ITEMS.length)],
      quantity: Math.floor(Math.random() * 5) + 1,
      unit_price: Math.round((5 + Math.random() * 45) * 100) / 100,
      created_at: new Date().toISOString(),
    },
  });
}

return results;
